src/api/customer/customer-serializer.ts

```diff
@@ -19,6 +23,71 @@ const createCustomerSerializer: CustomerCreationRequestHandler = (req, res, next
     next();
 }
 
+const paginationSerializer: GetCustomersAllRequestHandler = (req, res, next) => {
+    const paginationParams = req.query;
+    const pagination = {
+        offset: 0,
+        limit: 50,
+    }
+  
+    pagination.offset = paginationParams.offset ? Number(paginationParams.offset): pagination.offset;
+    pagination.limit = paginationParams.limit ? Number(paginationParams.limit) : pagination.limit;
+  
+    res.locals.pagination = pagination;
+  
+    next();
+}
+
+const getAllCustomersSerializer: GetCustomersAllRequestHandler = (req, res, next) => {
+    const customers = res.locals.getCustomer;
+    res.locals.customerToRespond = [];
+
+    customers.map((customer) => {
+        res.locals.customerToRespond.push({
+            uuid: customer.uuid,
+            name: customer.name,
+            contact: {
+                email: customer.email,
+                phone: customer.phone,
+            },
+            document: {
+                cpf: customer.cpf,
+                cnpj: customer.cnpj,
+            },
+            createdAt: customer.createdAt.toISOString(),
+            updatedAt: customer.updatedAt.toISOString(),
+        });
+    });
```

Comment:
Poderia fazer a atribuição diretamente, ao invés de criar um array vazio e popular posteriormente.

OBS: `Esta função (e demais partes do código) está semelhante a um código enviado por outro aluno.`
Talvez vocês tenham encontrado códigos similares na internet. :thinking:

```suggestion
    res.locals.customerToRespond = customers.map((customer) => ({
          uuid: customer.uuid,
          name: customer.name,
          contact: {
              email: customer.email,
              phone: customer.phone,
          },
          document: {
              cpf: customer.cpf,
              cnpj: customer.cnpj,
          },
          createdAt: customer.createdAt.toISOString(),
          updatedAt: customer.updatedAt.toISOString(),
      }));
```